com.microsoft.mixedreality.visualprofiler/VisualProfiler.cs

```diff
@@ -911,24 +942,31 @@ private void LateUpdate()
                 {
                     if (profilerGroup.ReadyToPresent())
                     {
-                        float milliseconds = profilerGroup.CalculateAverage() * 1e-6f;
+                        float value = profilerGroup.CalculateDisplayValue();
 
-                        if (WillDisplayedMillisecondsDiffer(profilerGroup.LastValuePresented, milliseconds, displayedDecimalDigits))
+                        if (WillDisplayedProfilerValueDiffer(profilerGroup.LastValuePresented, value, displayedDecimalDigits))
                         {
                             profilerGroup.HasEverPresented = true;
-                            profilerGroup.LastValuePresented = milliseconds;
+                            profilerGroup.LastValuePresented = value;
 
-                            float budget = TargetFrameTime * profilerGroup.BudgetPercentage;
-                            Color color = milliseconds <= budget ? targetFrameRateColor : missedFrameRateColor;
-                            MillisecondsToString(stringBuffer, displayedDecimalDigits, profilerGroup.Text, milliseconds, color, maxStringLength);
+                            Color color = Color.white;
+
+                            // Only apply frame time budget coloring to time profilers

```

Comment:
I could see a future where we have budgets for other units - but this is totally cool for v1!